### PR TITLE
[Enhancement]: add md link check github action

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Check dead links in documents at PR stage.

## Modification

Add link check in `lint` ci.

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RunningLeon** [comment]
@lvhan028 @AllentDan Hi, guys, any comment on this PR?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -27,6 +27,15 @@ jobs:
...
           python .github/scripts/check_index_rst.py docs/en/index.rst
           python .github/scripts/check_index_rst.py docs/zh_cn/index.rst
+      - name: Check markdown link
+        uses: gaurav-nelson/github-action-markdown-link-check@v1
+        with:
+          use-quiet-mode: 'yes'
+          use-verbose-mode: 'yes'
+#          check-modified-files-only: 'yes'
+          config-file: '.github/md-link-config.json'
+          folder-path: 'docs/en, docs/zh_cn'
```
Comment (AllentDan) [inline .github/workflows/lint.yml:37]: Does the tool only work for markdown links or beyond?

Comment (AllentDan) [reply .github/workflows/lint.yml:37]: I saw that even some wrong links in python files got detected.

Comment (RunningLeon) [reply .github/workflows/lint.yml:37]: It only checks the links in markdown.

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**